src/goto-instrument/concurrency.cpp

```diff
@@ -58,14 +58,14 @@ class concurrency_instrumentationt
   {
   public:
     typet type;
-    symbol_exprt array_symbol, w_index_symbol;
+    exprt array_symbol, w_index_symbol;
   };
```

Comment:
Can these ever be something that's not a symbol_exprt?
If initialisation is delayed, then perhaps optionalt<symbol_exprt> is the right way to say that.

Reply:
Well, the code is not particularly complete. These are actually never set at the moment.